GNOME 48 support

Thesola10 opened the pull request:
This fixes #1002 by restoring PaperWM functionality on the latest GNOME 48.

Contributions are welcome!

Thesola10 committed:
Updated test VM to GNOME 48

mehiel commented:
Tried this branch after setting metadata.json to support 48 version but unfortunately when I activate the extension breaks gnome which exists the current sessions and goes back to gdm to login.

Plyply99 commented:
> Tried this branch after setting metadata.json to support 48 version but unfortunately when I activate the extension breaks gnome which exists the current sessions and goes back to gdm to login.

I experienced exact same thing.

Thesola10 commented:
There are no changes yet, just a setup for GNOME 48 testing, hence the "contributions are welcome".
Also yeah I forgot to bump metadata.json

Thesola10 committed:
Bumped metadata for testing

amfern commented:
I get a crash when trying to use this branch.
[journal-paperwm.log](https://github.com/user-attachments/files/19401438/journal-paperwm.log)

amfern commented:
[96ae4ebf-5028-4dce-a82c-bd076da20b03.log](https://github.com/user-attachments/files/19403302/96ae4ebf-5028-4dce-a82c-bd076da20b03.log)
gdb back trace

Thesola10 committed:
Updated test VM to reduce build time

This change also removes GNOME Circle apps from the test VM.

azmeuk commented:
No crash for me, but navigation shortcuts are broken.

Thesola10 committed:
Fixed crash on Wayland

schrieveslaach commented:
> No crash for me, but navigation shortcuts are broken.

Same on my machine

Thesola10 committed:
Fixed navigator keybindings

Thesola10 commented:
@schrieveslaach @azmeuk just pushed a fix for navigation keybinds

azmeuk commented:
> @schrieveslaach @azmeuk just pushed a fix for navigation keybinds

I confirm this fixes the issues for me.

fermuch commented:
> @schrieveslaach @azmeuk just pushed a fix for navigation keybinds

I can confirm it is working for me too. Thanks!

Thesola10 commented on topbar.js:
```diff
@@ -72,9 +72,10 @@ export function enable (extension) {
     Main.panel.addToStatusArea('OpenPositionButton', openPositionButton, 3, 'left');
 
+    /* This causes a crash on GNOME 48
     Tiling.spaces.forEach(s => {
         s.workspaceLabel.clutter_text.set_font_description(menu.label.clutter_text.font_description);
-    });
+    });*/
```
Need to investigate effects on GNOME 47, most likely cosmetic.

Thesola10 replied:
Yep, seems unnecessary on modern GNOME.

jtaala replied:
Indeed, likely from previous gnome versions.  Have tested on 47 and doesn't look like needed (or at least couldn't notice any issues here).

thenameisluk commented:
just wanted to let you know
that dragging window using titlebar
is instantly crashing shell
```
Distribution: Debian GNU/Linux
GNOME Shell: 48.0
Display server: Wayland
PaperWM version: 48.0.0
Enabled extensions:
- gravatar@dsheeler.net
- gnomebedtime@ionutbortis.gmail.com
- caffeine@patapon.info
- weatheroclock@CleoMenezesJr.github.io
- custom-command-toggle@storageb.github.com
- ddterm@amezin.github.com
```
(on this branch)

Thesola10 commented:
@thenameisluk It seems to be an [upstream issue](https://gitlab.gnome.org/GNOME/mutter/-/issues/3960) somehow??? Apparently they [merged a fix](https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/4307) a few days ago

In the mean time you can disable cursor change by removing line 69 from `grab.js`, might block cursor change if version=48.0
```js
        global.display.set_cursor(Meta.Cursor.MOVE_OR_RESIZE_WINDOW);
```

schrieveslaach commented:
> @schrieveslaach @azmeuk just pushed a fix for navigation keybinds

Super + . and other keybindings work again.

Thesola10 committed:
Added exceptions for new cursor names in GNOME 48

Of course this wasn't mentioned in any release notes, and they just HAD
to throw away the old names. Combined with JS going "lol undefined" and
Clutter happily eating it and crashing, it's the perfect storm.

thenameisluk commented:
this only postponed the crash
it will work if you just grab it
but moving it down will again result in a crash

data-flux commented:
Fedora Silverblue 42 beta, gnome version 48rc
- Keyboard navigation works perfectly!
- moving tiled windows with the mouse still crashes.

Thesola10 commented:
Nevermind, found it: for some reason `Meta.Cursor.MOVE_OR_RESIZE_WINDOW` is undefined on GNOME 48, and of course Clutter responds by imploding on the spot.

~~Sure, GNOME, why don't you rename the cursors without a release note, and not bother handling the old names. Gotta love Javascript.~~

thenameisluk commented:
this fixed the crash
thx

jtaala reviewed (approved):
Nice work @Thesola10!

Now running this - working well.  Happy for this to be merged when you're ready.

jtaala commented:
@Thesola10, let me know if you're waiting on anything, otherwise I can merge and create a new release (on EGO as well).

Jay.

Thesola10 commented:
No, please go ahead, had other things to do today and forgot

jtaala commented:
> No, please go ahead, had other things to do today and forgot

No problems!
